### PR TITLE
[native] Add e2e test for insert into partitioned table

### PR DESCRIPTION
**Pull request by gggrace14**
Add a native e2e unit test to test insert into a partitioned table.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by gggrace14** [comment]
Velox/native worker doesn't support experimental-optimized-partition-update-serialization-enabled yet, which is SMILE serde and zstd compress. So setting experimental-optimized-partition-update-serialization-enabled to false for the unit test query runner. PrestoSparkQueryRunner is also shared by Java preso-spark. Let me know if I need to split the hive config setting for native worker and Java worker.

File: presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java [hunk 1]
```diff
@@ -369,7 +369,7 @@ public PrestoSparkQueryRunner(
         pluginManager.installPlugin(new HivePlugin("hive", Optional.of(metastore)));
 
         Map<String, String> properties = ImmutableMap.<String, String>builder()
-                .put("hive.experimental-optimized-partition-update-serialization-enabled", "true")
+                .put("hive.experimental-optimized-partition-update-serialization-enabled", "false")
```
Comment (mbasmanova) [inline presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:372]: I think this is used for Java tests as well. Can we change buildSessionForTableWrite instead?

Comment (miaoever) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java:372]: @gggrace14 better to change https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java#L136 which only used by pos native to pass in the required configs.

Comment (gggrace14) [reply presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java]: > @gggrace14 we can change https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/spark/PrestoSparkNativeQueryRunnerUtils.java#L136 which only used by pos native to pass in the required configs.

Hi @miaoever , considered setting the session property in PrestoSparkNativeQueryRunnerUtils for native tests at first, but it calls PrestoSparkQueryRunner where the session property is set to true currently. Thus an error of duplicate property setting is raised.

I'm thus taking the approach that Masha suggests to set the session in the unit tests. Looks a more straightforward way.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 2]
```diff
@@ -866,6 +866,38 @@ public void testCreateUnpartitionedTableAsSelect()
...
+
+            // Test failure on insert into existing partitions.
+            assertQueryFails(writeSession, String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName),
+                    ".*Cannot insert into an existing partition of Hive table: regionkey=.*/nationkey=.*");
+
+            // Test insert into existing partitions if insert_existing_partitions_behavior is set to OVERWRITE.
+            Session overwriteSession = Session.builder(writeSession)
+                    .setCatalogSessionProperty("hive", "insert_existing_partitions_behavior", "OVERWRITE")
+                    .build();
+            getQueryRunner().execute(overwriteSession, String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
```
Comment (mbasmanova) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:893]: nit: perhaps, modify the query to insert different data, say name || '.test', to make sure the overwrite actually happens

**Review by mbasmanova — COMMENTED** [review]
@gggrace14 Ge, thank you for adding this test. Looks good overall % commit message needs updating. Please, remember to add [native] prefix to commit title.

**Comment by gggrace14** [comment]
Hi @mbasmanova , this PR is ready again. Revised per comment left and have the green tests now.

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Pull request merged**